subsystem/cluster_test.go

```diff
@@ -3043,6 +3045,72 @@ var _ = Describe("cluster install, with default network params", func() {
 		})
 	})
 })
+var _ = Describe("Update cluster with OLM operators", func() {
+	var (
+		clusterID strfmt.UUID
+		hosts     []*models.Host
+	)
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	BeforeEach(func() {
+		ctx := context.Background()
+		clstrID, err := registerCluster(ctx, userBMClient, "test-cluster", pullSecret)
+		Expect(err).NotTo(HaveOccurred())
+		clusterID = clstrID
+		hosts = registerHostsAndSetRolesDHCP(clusterID, 3)
+		inventory := *validHwInfo
+		// The hosts need to have enough memory for hosting OCS
+		inventory.Memory = &models.Memory{PhysicalBytes: int64(40 * units.GiB), UsableBytes: int64(40 * units.GiB)}
+		for _, disk := range inventory.Disks {
+			// The disks need to be too small for OCP with OCS
+			disk.SizeBytes = conversions.GbToBytes(120)
+		}
+		for i, host := range hosts {
+			generateHWPostStepReply(ctx, host, &inventory, fmt.Sprintf("host-%v", i))
+		}
+
+	})
+
+	It("should change disk eligibility", func() {
+		By("sanity check: having eligible disks")
+		cluster := getCluster(clusterID)
+		for _, host := range cluster.Hosts {
+			Expect(*host.Status).To(BeEquivalentTo(models.HostStatusKnown))
+			inventory, err := hostutil.UnmarshalInventory(host.Inventory)
+			Expect(err).ToNot(HaveOccurred())
+			for _, disk := range inventory.Disks {
+				Expect(disk.InstallationEligibility.Eligible).To(BeTrue())
+				Expect(disk.InstallationEligibility.NotEligibleReasons).To(BeEmpty())
+			}
+		}
+
+		By("enabling OCS operator - increasing disk requirements")
+		_, err := userBMClient.Installer.UpdateCluster(context.TODO(), &installer.UpdateClusterParams{
```

Comment:
I would suggest checking the update result this will validate that UI will get the right reply

Reply:
Done.